src/cargo/core/compiler/compilation.rs

```diff
@@ -372,7 +372,7 @@ impl<'cfg> Compilation<'cfg> {
 /// Prepares a rustc_tool process with additional environment variables
 /// that are only relevant in a context that has a unit
 fn fill_rustc_tool_env(mut cmd: ProcessBuilder, unit: &Unit) -> ProcessBuilder {
-    if unit.target.is_bin() {
+    if unit.target.is_executable() {
```

Comment:
Aside: I feel like this should be `is_runnable` (ie `cargo run`).  I saw the name and thought it included tests and was about to comment on the documentation when I decided to check the code and it turns out `is_executable` is actually only for bins and examples.